linalg/hypre.cpp

```diff
@@ -1293,8 +1293,14 @@ void HypreParMatrix::Read(MPI_Comm comm, const char *fname)
    Destroy();
    Init();
 
-   HYPRE_Int base_i, base_j;
-   hypre_ParCSRMatrixReadIJ(comm, fname, &base_i, &base_j, &A);
+   HYPRE_IJMatrix A_ij;
```

Comment:
This is braking backward compatibility -- the `ParCSR` and `IJ` formats are different.

Maybe it is best to split them in separate functions: `ReadParCSR` and `ReadIJ` and have `Read` call `ReadParCSR` to maintain the old behavior?